core/src/main/java/org/apache/brooklyn/enricher/stock/PercentageEnricher.java

```diff
@@ -0,0 +1,127 @@
+    protected AttributeSensor<? extends Number> sourceTotalSensor;
+    protected AttributeSensor<Double> targetSensor;
+    protected Entity producer;
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    @Override
+    public void setEntity(EntityLocal entity) {
+        super.setEntity(entity);
+
+        this.sourceCurrentSensor = getConfig(SOURCE_CURRENT_SENSOR);
+        this.sourceTotalSensor = getConfig(SOURCE_TOTAL_SENSOR);
+        this.targetSensor = getConfig(TARGET_SENSOR);
+        this.producer = getConfig(PRODUCER) == null ? entity : getConfig(PRODUCER);
+
+        if (sourceCurrentSensor == null) {
+            throw new IllegalArgumentException("Enricher " + JavaClassNames.simpleClassName(this) + " has no " + SOURCE_CURRENT_SENSOR.getName());
+        }
+        if (sourceTotalSensor == null) {
+            throw new IllegalArgumentException("Enricher " + JavaClassNames.simpleClassName(this) + " has no " + SOURCE_TOTAL_SENSOR.getName());
+        }
+        if (targetSensor == null) {
+            throw new IllegalArgumentException("Enricher " + JavaClassNames.simpleClassName(this) + " has no " + TARGET_SENSOR.getName());
+        }
+
+        if (targetSensor.equals(sourceCurrentSensor)) {
+            throw new IllegalArgumentException("Enricher " + JavaClassNames.simpleClassName(this) + " detect cycle with " + SOURCE_CURRENT_SENSOR.getName());
+        }
+        if (targetSensor.equals(sourceTotalSensor)) {
+            throw new IllegalArgumentException("Enricher " + JavaClassNames.simpleClassName(this) + " detect cycle with " + SOURCE_TOTAL_SENSOR.getName());
+        }
+
+        subscriptions().subscribe(MutableMap.of("notifyOfInitialValue", true), producer, sourceCurrentSensor, this);
+        subscriptions().subscribe(MutableMap.of("notifyOfInitialValue", true), producer, sourceTotalSensor, this);
+
+    }
+
+    @Override
+    public void onEvent(SensorEvent<Number> event) {
+        Number current = producer.sensors().get(sourceCurrentSensor);
+        Number total = producer.sensors().get(sourceTotalSensor);
+        Double result = null;
+
+        if (current == null) {
+            LOG.debug("Current is null, returning");
+            return;
+        }
+
+        if (total == null || total.equals(0)) {
+            LOG.debug("total {" + total + "} is null or zero, returning");
+            return;
+        }
+
+        Double currentDouble = current.doubleValue();
+        Double totalDouble = total.doubleValue();
+
+        if (currentDouble > totalDouble) {
+            LOG.debug("Current is greater than total, returning");
+            return;
+        }
+
+        if (currentDouble < 0 || totalDouble < 0) {
+            LOG.debug("Current {"+currentDouble+"}  or total {"+totalDouble+"} is negative, returning");
+            return;
+        }
+
+        if (current.equals(0)) {
+            LOG.debug("current is zero, setting percent to zero");
+            result = 0d;
```

Comment:
This gives the same result as `0d / totalDouble` i.e. `0d` so don't need the check, just always do the division.